crates/bevy_ui/src/focus.rs

```diff
@@ -166,12 +204,352 @@ pub fn ui_focus_system(
             FocusPolicy::Pass => { /* allow the next node to be hovered/clicked */ }
         }
     }
-    // reset lower nodes to None
-    for (_entity, _focus_policy, interaction, _) in moused_over_z_sorted_nodes {
-        if let Some(mut interaction) = interaction {
-            if *interaction != Interaction::None {
-                *interaction = Interaction::None;
+}
+
+trait CursorResource: Resource {
+    fn get_cursor_position(&self) -> Option<Vec2>;
+}
+
+impl CursorResource for Windows {
+    fn get_cursor_position(&self) -> Option<Vec2> {
+        self.get_primary()
+            .and_then(|window| window.cursor_position())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use rstest::*;
```

Comment:
I don't think that this adds enough value to pull in a testing library on a one-off basis. These tests should be feasibel to rewrite with a helper method instead.

Reply:
I kinda handcoded it myself now

Reply:
Excellent, thanks!